src/software/multithreading/observer.tpp

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
 template <typename T>
-Observer<T>::Observer(size_t buffer_size) : buffer(buffer_size)
+Observer<T>::Observer(size_t buffer_size)
+    : buffer(buffer_size), time_buffer(TIME_BUFFER_SIZE)
 {
 }
 
 template <typename T>
 void Observer<T>::receiveValue(T val)
 {
+    time_buffer.push_back(std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch()));
```

Comment:
super nit: It's generally better to use the `monotonic_clock` for measuring local time differences / durations, because that clock only moves "forwards" in time and isn't subject to other variations.

Reply:
changed